src/control/cmd/dmg/pretty/storage_nvme.go

```diff
@@ -227,6 +236,18 @@ func PrintNvmeControllers(controllers storage.NvmeControllers, out io.Writer, op
 		row[fwTitle] = ctrlr.FwRev
 		row[socketTitle] = fmt.Sprint(ctrlr.SocketID)
 		row[capacityTitle] = humanize.Bytes(ctrlr.Capacity())
+		roles := "N/A"
+		rank := "None"
+		// Assumes that all SMD devices on a controller have the same roles and rank.
+		if len(ctrlr.SmdDevices) > 0 {
+			sd := ctrlr.SmdDevices[0]
+			roles = fmt.Sprintf("%s", &sd.Roles)
```

Comment:
minor: Seems a little weird to use `fmt.Sprintf()` instead of calling `sd.Roles.String()`.

Reply:
fixed